File: app/src/main/java/android/net/NetworkTemplate.java
package android.net;

import io.github.francoiscampbell.xposeddatausage.StubClassNotShadowedError;

/**
 * Created by francois on 16-03-15.
 */
public abstract class NetworkTemplate {
    //repeat this string in all classes because some but not all may be shadowed by the system

    //These implementations don't matter, the actual ones executed will be the system's
    public static NetworkTemplate buildTemplateMobileAll(String subscriberId) {
        throw new StubClassNotShadowedError();
    }

    public static NetworkTemplate buildTemplateWifi() {
        throw new StubClassNotShadowedError();
    }

    public static NetworkTemplate normalize(NetworkTemplate template, String[] merged) {
        throw new StubClassNotShadowedError();
    }
}

File: app/src/main/kotlin/io/github/francoiscampbell/xposeddatausage/stubs/android/net/NetworkTemplate.kt
package io.github.francoiscampbell.xposeddatausage.stubs.android.net

import de.robv.android.xposed.XposedHelpers

/**
 * Created by francois on 16-03-13.
 */
object NetworkTemplate {
    val stubClass = XposedHelpers.findClass("android.net.NetworkTemplate", null)

    fun buildTemplateMobileAll(subscriberId: String) = XposedHelpers.callStaticMethod(stubClass, "buildTemplateMobileAll", subscriberId)
}